app-crypt/certbot/certbot-3.2.0-r100.ebuild

```diff
@@ -0,0 +1,270 @@
+		>=dev-python/google-auth-2.16.0[${PYTHON_USEDEP}]
+	)
+	certbot-dns-linode? (
+		>=dev-python/dns-lexicon-3.14.1[${PYTHON_USEDEP}]
+	)
+	certbot-dns-luadns? (
+		>=dev-python/dns-lexicon-3.14.1[${PYTHON_USEDEP}]
+	)
+	certbot-dns-nsone? (
+		>=dev-python/dns-lexicon-3.14.1[${PYTHON_USEDEP}]
+	)
+	certbot-dns-ovh? (
+		>=dev-python/dns-lexicon-3.15.1[${PYTHON_USEDEP}]
+	)
+	certbot-dns-rfc2136? (
+		>=dev-python/dnspython-2.6.1[${PYTHON_USEDEP}]
+	)
+	certbot-dns-route53? (
+		>=dev-python/boto3-1.15.15[${PYTHON_USEDEP}]
+	)
+	certbot-dns-sakuracloud? (
+		>=dev-python/dns-lexicon-3.14.1[${PYTHON_USEDEP}]
+	)
+	certbot-nginx? (
+		>=dev-python/pyopenssl-25.0.0[${PYTHON_USEDEP}]
+		>=dev-python/pyparsing-2.4.7[${PYTHON_USEDEP}]
+	)
+	selinux? ( sec-policy/selinux-certbot )
+"
+# RDEPEND+="
+# 	>=dev-python/requests-toolbelt-0.3.0[${PYTHON_USEDEP}] # @TODO is still necessary?
+# 	certbot-dns-cloudflare? (
+# 		# Available in GURU
+# 		>=dev-python/cloudflare-2.19[${PYTHON_USEDEP}]
+# 		<dev-python/cloudflare-2.20[${PYTHON_USEDEP}]
+# 	)
+# 	certbot-dns-digitalocean? (
+# 		# Available in GURU
+# 		>=dev-python/digitalocean-1.11[${PYTHON_USEDEP}]
+# 	)
+# "
+
+distutils_enable_sphinx docs \
+	dev-python/sphinx-rtd-theme
+distutils_enable_tests pytest
+
+src_prepare() {
+	local S_BACKUP="${S}"
+	local certbot_dirs=()
+	for base in "${CERTBOT_BASE[@]}"; do
+		certbot_dirs+=("${base}")
+	done
+	for module in "${CERTBOT_MODULES_EXTRA[@]}"; do
+		use "certbot-${module}" \
+			&& certbot_dirs+=("certbot-${module}")
+	done
+
+	for dir in "${certbot_dirs[@]}"; do
+		S="${WORKDIR}/${P}/${dir}"
+		pushd "${S}" > /dev/null || die
+		distutils-r1_src_prepare
+		popd > /dev/null || die
+	done
+
+	# Restaure S
+	S="${S_BACKUP}"
+}
+
+src_configure() {
+	local S_BACKUP="${S}"
+	local certbot_dirs=()
+	for base in "${CERTBOT_BASE[@]}"; do
+		certbot_dirs+=("${base}")
+	done
+	for module in "${CERTBOT_MODULES_EXTRA[@]}"; do
+		use "certbot-${module}" \
+			&& certbot_dirs+=("certbot-${module}")
+	done
+
+	for dir in "${certbot_dirs[@]}"; do
+		S="${WORKDIR}/${P}/${dir}"
+		pushd "${S}" > /dev/null || die
+		distutils-r1_src_configure
+		popd > /dev/null || die
+	done
+
+	# Restaure S
+	S="${S_BACKUP}"
+}
+
+src_compile() {
+	local S_BACKUP="${S}"
+	local certbot_dirs=()
+	for base in "${CERTBOT_BASE[@]}"; do
+		certbot_dirs+=("${base}")
+	done
+	for module in "${CERTBOT_MODULES_EXTRA[@]}"; do
+		use "certbot-${module}" \
+			&& certbot_dirs+=("certbot-${module}")
+	done
+
+	for dir in "${certbot_dirs[@]}"; do
+		S="${WORKDIR}/${P}/${dir}"
+		pushd "${S}" > /dev/null || die
+		distutils-r1_src_compile
+		popd > /dev/null || die
+	done
+
+	# Restaure S
+	S="${S_BACKUP}"
+}
+
+src_test() {
+	local S_BACKUP="${S}"
+	local certbot_dirs=()
+	for base in "${CERTBOT_BASE[@]}"; do
+		certbot_dirs+=("${base}")
+	done
+	for module in "${CERTBOT_MODULES_EXTRA[@]}"; do
+		use "certbot-${module}" \
+			&& certbot_dirs+=("certbot-${module}")
+	done
+
+	python_test() {
```

Comment:
We usually define this out-of-line (so not inside of `src_test`).

Reply:
Moved.